packages/scripts/src/addOrbitChain/transforms.ts

```diff
@@ -0,0 +1,382 @@
+    throw new Error(
+      `Invalid image extension '${fileExtension}'. Expected: ${SUPPORTED_IMAGE_EXTENSIONS.join(
+        ", "
+      )}.`
+    );
+  }
+
+  const imageSavePath = `images/${fileName}.${fileExtension}`;
+  const fullSavePath = `../../arb-token-bridge-ui/public/${imageSavePath}`;
+
+  // Create directories if they don't exist
+  const dirs = fullSavePath.split("/").slice(0, -1).join("/");
+  if (!fs.existsSync(dirs)) {
+    fs.mkdirSync(dirs, { recursive: true });
+  }
+
+  if (fs.existsSync(fullSavePath)) {
+    warning(
+      `${fileName} already exists at '${imageSavePath}'. Using the existing image.`
+    );
+    return `/${imageSavePath}`;
+  }
+
+  const response = await axios.get(urlOrPath, { responseType: "arraybuffer" });
+  const imageBuffer = Buffer.from(response.data);
+
+  // Check if the file already exists in the repository
+  let sha: string | undefined;
+  try {
+    const { data } = await getContent(branchName, imageSavePath);
+
+    if ("sha" in data) {
+      sha = data.sha;
+    }
+  } catch (error) {
+    // File doesn't exist, which is fine
+  }
+
+  await updateContent(branchName, imageSavePath, imageBuffer, sha);
+
+  return `/${imageSavePath}`;
+};
+
+export const transformIncomingDataToOrbitChain = (
+  chainData: IncomingChainData,
+  chainLogoPath: string,
+  nativeTokenLogoPath?: string
+): OrbitChain => {
+  const parentChainId = parseInt(chainData.parentChainId, 10);
+  const isTestnet = TESTNET_PARENT_CHAIN_IDS.includes(parentChainId);
+
+  return {
+    chainId: parseInt(chainData.chainId, 10),
+    confirmPeriodBlocks: parseInt(chainData.confirmPeriodBlocks, 10),
+    ethBridge: {
+      bridge: chainData.bridge,
+      inbox: chainData.inbox,
+      outbox: chainData.outbox,
+      rollup: chainData.rollup,
+      sequencerInbox: chainData.sequencerInbox,
+    },
+    nativeToken: chainData.nativeTokenAddress,
+    explorerUrl: chainData.explorerUrl,
+    rpcUrl: chainData.rpcUrl,
+    isArbitrum: true,
+    isCustom: true,
+    isTestnet,
+    name: chainData.name,
+    slug: nameToSlug(chainData.name),
+    parentChainId,
+    partnerChainIDs: [],
+    retryableLifetimeSeconds: 604800,
+    tokenBridge: {
+      parentCustomGateway: chainData.parentCustomGateway,
+      parentErc20Gateway: chainData.parentErc20Gateway,
+      parentGatewayRouter: chainData.parentGatewayRouter,
+      parentMulticall: chainData.parentMulticall,
+      parentProxyAdmin: chainData.parentProxyAdmin,
+      parentWeth: chainData.parentWeth,
+      parentWethGateway: chainData.parentWethGateway,
+      childCustomGateway: chainData.childCustomGateway,
+      childErc20Gateway: chainData.childErc20Gateway,
+      childGatewayRouter: chainData.childGatewayRouter,
+      childMulticall: chainData.childMulticall,
+      childProxyAdmin: chainData.childProxyAdmin,
+      childWeth: chainData.childWeth,
+      childWethGateway: chainData.childWethGateway,
+    },
+    nitroGenesisBlock: 0,
```

Comment:
`nitroGenesisBlock`
`nitroGenesisL1Block`
`depositTimeout`
`blockTime`
not used